docs/user-manual/en/address-model.md

```diff
@@ -0,0 +1,585 @@
+
+```xml
+<configuration ...>
+  <core ...>
+    ...
+    <address name="foo.orders">
+      <anycast>
+        <queue name="orders"/>
+      </anycast>
+      <multicast/>
+    </address>
+  </core>
+</configuration>
+```
+
+## How to filter messages
+
+Apache ActiveMQ Artemis supports the ability to filter messages using Apache Artemis [Filter Expressions](#filter-expressions).
+
+Filters can be applied in two places, on a queue and on a consumer.
+
+### Queue Filter
+
+When a filter is applied to a queue, messages are filter before they sent to the queue.  To add a queue filter use the
+filter element when configuring a queue.  Open up the broker.xml and add an address with a queue, using the filter element
+to configure a filter on this queue.
+
+```xml
+    <address name="filter">
+       <queue name=filter">
+          <filter string="color='red'"/>
+        </queue>
+    </address>
+```
+
+The filter defined above ensures that only messages with an attribute "color='red'" is sent to this queue.
+
+### Consumer Filters
+
+Consumer filters are applied after messages have reached a queue and are defined using the appropriate client APIs.  The
+follow JMS example shows how to consumer filters work.
+
+1. Define an address with a single queue, with no filter applied.
+
+```xml
+    <address name="filter">
+       <queue name=filter">
+          <filter string="color='red'"/>
+        </queue>
+    </address>
+```
+
+```java
+  ...
+  // Send some messages
+  for (int i = 0; i < 3; i ++) {
+    TextMessage redMessage = senderSession.createTextMessage("Red");
+    redMessage.setStringProperty("color", "red");
+    producer.send(redMessage)
+        
+    TextMessage greenMessage = senderSession.createTextMessage("Green");
+    greenMessage.setStringProperty("color", "green");
+    producer.send(greenMessage)
+  }
+```
+
+At this point the queue would have 6 messages: red,green,red,green,red,green
+         
+```java
+  MessageConsumer redConsumer = redSession.createConsumer(queue, "color='red'");
+```
+
+The redConsumer has a filter that only matches "red" messages.  The redConsumer will receive 3 messages.
+
+```
+red, red, red
+```
+
+The resulting queue would now be
+
+```
+green, green, green
+```
+
+## Creating and Deleting Addresses and Queues Automatically
+
+You can configure Apache ActiveMQ Artemis  to automatically create addresses and then delete them when they are no longer in use. This saves you from having to preconfigure each address before a client can connect to it. Automatic creation and deletion is configured on a per address basis and is controlled by following
+
+| Parameter | Description |
+---------------------------
+| auto-create-addresses | When set to true, the broker will create the address requested by the client if it does not exist already. The default is true.|
+| auto-delete-addresses | When set to true, the broker will be delete the address once all of it’s queues have been deleted. The default is true |
```

Comment:
It may be important to note that only auto-created addresses will get auto-deleted.